cmake/CMakeDetermineMetalCompiler.cmake

```diff
@@ -9,7 +9,7 @@ if(NOT CMAKE_Metal_COMPILER_NAMES)
     set(CMAKE_Metal_COMPILER_NAMES metal)
 endif()
 
-if("${CMAKE_GENERATOR}" STREQUAL "Xcode")
+if("${CMAKE_GENERATOR}" STREQUAL "Xcode" OR CMAKE_VERSION VERSION_GREATER_EQUAL "4.0")
```

Comment:
I think we might need a check for Darwin here too, because Apple does provide the Metal compiler as a download for Windows and `xcrun --find` won't work there